src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -422,7 +419,7 @@ private static Details getDetailsForTryCatchFinally(DetailAST ast) {
                     nextToken = getNextToken(ast);
                 }
             }
-            else if (tokenType == TokenTypes.LITERAL_CATCH) {
+            else {
```

Comment:
Catch and finally are handled and computed exactly the same thanks to the already existing `null` check.